python/lsst/ts/hexrotcomm/command_telemetry_client.py

```diff
@@ -168,9 +171,12 @@ def __init__(
         # by the data type of Command.counter.
         self._command_counts = dict()
 
-        # Task the user can set to an asyncio.Future()
-        # in order to detect when the next telemetry is read.
-        # Only intended for unit tests.
+        # Task set to None when config is first seen and handled by
+        # config_callback, or the exception if config_callback raises.
+        self.configured_task = asyncio.Future()
```

Comment:
Why do not you use `make_done_future()` in `ts_utils`? In addition, why it is `self.configured_task` instead of `self._configured_task` because you have the `self._telemetry_task` and other tasks already?

Reply:
I am using asyncio.Future instead of utils.make_done_future() because this task should not be marked done until the first config is seen after connecting. It's not a really big deal either way -- the user could create a new one before connecting. But this way the user does not need to do anything extra.

It is `configured_task` instead of `_configured_task` because it is part of the public API. By comparison `_telemetry_task` is a hidden detail of the `next_telemetry` method. I thought about making them fully symmetric and having a `next_config` method, but it turns out that's very much the wrong thing to do. Telemetry comes out frequently so waiting for the next one is very reasonable. But config is only published once at connect time and them possibly never again.

I also thought about naming it `initial_config_task` but with the current naming the user can reset it to a new Future and wait for a new config if they want (we have no use case for that except possibly in unit tests).